### PR TITLE
Modification of the variant annotation schema

### PR DESCRIPTION
**Pull request by ireneisdoomed**
This PR includes changes to the variant annotation schema to remove redundant information. 
The changes described here align with those made in the Genetics schemas revision spreadsheet for the dataset `input.variant-annotation`. Specifically in the `to_change_v2` sheet.

- Drop of the field `locus`. Struct containing chrom and pos. This info was not used downstream and is found in the `chrom` and `pos` fields.
- Drop of the field `locus_GRCh38`. Duplication of `locus`.
- Drop of the field `locus`. Array of size 2 containing the ref and alt alleles. This info was not used downstream and is found in the `ref` and `alt` fields.
- Drop of `chrom_b38`. Same content as `chrom`
- Rename `pos_b38` to `pos`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@ dist/
 src/config.yaml
 **/__pycache__/
 outputs/
+src/etl/.DS_Store
```
Comment (d0choa) [inline .gitignore:5]: What about `**/.DS_Store`?

File: src/run_variant_annotation.py [hunk 2]
```diff
@@ -164,14 +181,14 @@ def main(cfg: DictConfig) -> None:
             ),
         )
         # Drop unused column:
-        .drop("locus", "alleles", "transcript_consequences", "af")
+        .drop("locus", "alleles", "transcript_consequences", "alleleFrequencies")
     )
 
-    validate_df_schema(variants, "variant_annotation.json")
+    # validate_df_schema(variants, "variant_annotation.json")
```
Comment (d0choa) [inline src/run_variant_annotation.py:187]: we need to update the schema and uncomment this before merging

Comment (ireneisdoomed) [reply src/run_variant_annotation.py:187]: yes yes

File: pyproject.toml [hunk 3]
```diff
@@ -56,6 +56,10 @@ pytest = "^7.1.3"
 pytest-cov = "^4.0.0"
 pytest-sugar = "^0.9.5"
 
+
+[tool.poetry.group.dev.dependencies]
+pydocstyle = "^6.1.1"
```
Comment (d0choa) [inline pyproject.toml:61]: We use `flake8-docstrings` which uses pydocstyle as a flake8 plugin. We shouldn't need this dependency

**Comment by ireneisdoomed** [comment]
image

After many iterations, I am satisfied with the current state of the changes.
The variant annotation job runs in 2h and I am able to query it and use it in further scripts in a reasonable time.
In the end, the best way I found to optimise it was to **coalesce all the data in 400 partitions and write it partitioned by chromosome**.
```
# location gs://genetics_etl_python_playground/XX.XX/output/python_etl/parquet/variant_annotation_broadcast_part
# schema
 |-- id: string (nullable = true)
 |-- position: integer (nullable = true)
 |-- referenceAllele: string (nullable = true)
 |-- alternateAllele: string (nullable = true)
 |-- chromosomeB37: string (nullable = true)
 |-- positionB37: integer (nullable = true)
 |-- alleleType: string (nullable = true)
 |-- rsIds: array (nullable = true)
 |    |-- element: string (containsNull = true)
 |-- alleleFrequencies: array (nullable = true)
 |    |-- element: struct (containsNull = true)
 |    |    |-- alleleFrequency: double (nullable = true)
 |    |    |-- populationName: string (nullable = true)
 |-- cadd: struct (nullable = true)
 |    |-- phred: float (nullable = true)
 |    |-- raw: float (nullable = true)
 |-- vep: struct (nullable = true)
 |    |-- mostSevereConsequence: string (nullable = true)
 |    |-- motifFeatureConsequences: array (nullable = true)
 |    |    |-- element: struct (containsNull = true)
 |    |    |    |-- allele_num: integer (nullable = true)
 |    |    |    |-- consequence_terms: array (nullable = true)
 |    |    |    |    |-- element: string (containsNull = true)
 |    |    |    |-- high_inf_pos: string (nullable = true)
 |    |    |    |-- impact: string (nullable = true)
 |    |    |    |-- minimised: integer (nullable = true)
 |    |    |    |-- motif_feature_id: string (nullable = true)
 |    |    |    |-- motif_name: string (nullable = true)
 |    |    |    |-- motif_pos: integer (nullable = true)
 |    |    |    |-- motif_score_change: double (nullable = true)
 |    |    |    |-- strand: integer (nullable = true)
 |    |    |    |-- variant_allele: string (nullable = true)
 |    |-- regulatoryFeatureConsequences: array (nullable = true)
 |    |    |-- element: struct (containsNull = true)
 |    |    |    |-- allele_num: integer (nullable = true)
 |    |    |    |-- biotype: string (nullable = true)
 |    |    |    |-- consequence_terms: array (nullable = true)
 |    |    |    |    |-- element: string (containsNull = true)
 |    |    |    |-- impact: string (nullable = true)
 |    |    |    |-- minimised: integer (nullable = true)
 |    |    |    |-- regulatory_feature_id: string (nullable = true)
 |    |    |    |-- variant_allele: string (nullable = true)
 |    |-- transcriptConsequences: array (nullable = true)
 |    |    |-- element: struct (containsNull = true)
 |    |    |    |-- allele_num: integer (nullable = true)
 |    |    |    |-- amino_acids: string (nullable = true)
 |    |    |    |-- appris: string (nullable = true)
 |    |    |    |-- biotype: string (nullable = true)
 |    |    |    |-- canonical: integer (nullable = true)
 |    |    |    |-- ccds: string (nullable = true)
 |    |    |    |-- cdna_start: integer (nullable = true)
 |    |    |    |-- cdna_end: integer (nullable = true)
 |    |    |    |-- cds_end: integer (nullable = true)
 |    |    |    |-- cds_start: integer (nullable = true)
 |    |    |    |-- codons: string (nullable = true)
 |    |    |    |-- consequence_terms: array (nullable = true)
 |    |    |    |    |-- element: string (containsNull = true)
 |    |    |    |-- distance: integer (nullable = true)
 |    |    |    |-- domains: array (nullable = true)
 |    |    |    |    |-- element: struct (containsNull = true)
 |    |    |    |    |    |-- db: string (nullable = true)
 |    |    |    |    |    |-- name: string (nullable = true)
 |    |    |    |-- exon: string (nullable = true)
 |    |    |    |-- gene_id: string (nullable = true)
 |    |    |    |-- gene_pheno: integer (nullable = true)
 |    |    |    |-- gene_symbol: string (nullable = true)
 |    |    |    |-- gene_symbol_source: string (nullable = true)
 |    |    |    |-- hgnc_id: string (nullable = true)
 |    |    |    |-- hgvsc: string (nullable = true)
 |    |    |    |-- hgvsp: string (nullable = true)
 |    |    |    |-- hgvs_offset: integer (nullable = true)
 |    |    |    |-- impact: string (nullable = true)
 |    |    |    |-- intron: string (nullable = true)
 |    |    |    |-- lof: string (nullable = true)
 |    |    |    |-- lof_flags: string (nullable = true)
 |    |    |    |-- lof_filter: string (nullable = true)
 |    |    |    |-- lof_info: string (nullable = true)
 |    |    |    |-- minimised: integer (nullable = true)
 |    |    |    |-- polyphen_prediction: string (nullable = true)
 |    |    |    |-- polyphen_score: double (nullable = true)
 |    |    |    |-- protein_end: integer (nullable = true)
 |    |    |    |-- protein_start: integer (nullable = true)
 |    |    |    |-- protein_id: string (nullable = true)
 |    |    |    |-- sift_prediction: string (nullable = true)
 |    |    |    |-- sift_score: double (nullable = true)
 |    |    |    |-- strand: integer (nullable = true)
 |    |    |    |-- swissprot: string (nullable = true)
 |    |    |    |-- transcript_id: string (nullable = true)
 |    |    |    |-- trembl: string (nullable = true)
 |    |    |    |-- tsl: integer (nullable = true)
 |    |    |    |-- uniparc: string (nullable = true)
 |    |    |    |-- variant_allele: string (nullable = true)
 |-- filters: array (nullable = true)
 |    |-- element: string (containsNull = true)
 |-- chromosome: string (nullable = true)
```

The schema reflects the changes proposed in the sheet `variant_changes_v2` of the [schemas spreadsheet](https://docs.google.com/spreadsheets/d/17Rjyv6YB5iXk0831gGvUNLgTaM2_jjTmp517Xg7iWSU/edit#gid=441156660).

Regarding **tests**: I didn't see where to put them. All the significant transformations happen within the Hail context.
Regarding **docs**: I've added them in the "variant" page, along with the variant index documentation (upcoming PR).

**Review by DSuveges — APPROVED** [review]

**Pull request merged**